File: c++/ResponsibilityPattern/Boss.cpp
#include "stdafx.h"
#include "Boss.h"


Boss::Boss(string name) : Handler(name)
{
	this->name = name;
}


void Boss::HandeRequest(Request request)
{
	cout<< this->name << "批准了" << request.name + "的" << request.days << "天假期." << endl;
}

File: csharp/DesignPattern.Responsibility/Boss.cs

using System;
namespace DesignPattern.Responsibility
{
    public class Boss : Handler
    {
        public Boss(string name)
            : base(name)
        {

        }

        public override void HandeRequest(Request request)
        {
            Console.WriteLine(this.Name + "批准了" + request.Name + "的" + request.Days + "天假期.");
        }
    }
}
